src/Symfony/Bundle/FeatureToggleBundle/DataCollector/FeatureCheckerDataCollector.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FeatureToggleBundle\DataCollector;
+
+use Symfony\Component\FeatureToggle\Feature;
+use Symfony\Component\FeatureToggle\FeatureCollection;
+use Symfony\Component\FeatureToggle\Strategy\StrategyInterface;
+use Symfony\Component\FeatureToggle\StrategyResult;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpKernel\DataCollector\LateDataCollectorInterface;
+use Symfony\Component\VarDumper\Caster\ClassStub;
+use Symfony\Component\VarDumper\Cloner\Data;
+
+/**
+ * @phpstan-type FeatureType array{
+ *     default: bool,
+ *     description: string,
+ *     strategy: StrategyInterface,
+ * }
+ * @phpstan-type ToggleType array{
+ *     feature: string,
+ *     result: bool|null,
+ *     computes: array<string, ComputeType>,
+ * }
+ * @phpstan-type ComputeType array{
+ *     strategyId: string,
+ *     strategyClass: string,
+ *     level: int,
+ *     result: StrategyResult|null,
+ * }
+ *
+ * @property Data|array{
+ *     features: array<string, FeatureType>,
+ *     toggles: array<string, ToggleType>,
+ * } $data
+ */
+final class FeatureCheckerDataCollector extends DataCollector implements LateDataCollectorInterface
+{
+    /** @var \SplStack<string> */
+    private \SplStack $currentToggle;
+
+    /** @var \SplStack<string> */
+    private \SplStack $currentCompute;
+
+    public function __construct(
+        private readonly FeatureCollection $featureCollection,
+    ) {
+        $this->data = ['features' => [], 'toggles' => []];
+        $this->currentToggle = new \SplStack();
+        $this->currentCompute = new \SplStack();
+    }
+
+    public function collect(Request $request, Response $response, \Throwable $exception = null): void
+    {
+        foreach ($this->featureCollection as $feature) {
```

Comment:
Should we really iterate over all registered features to collect them ? Those don't represent what happens during that request (and this will force to instantiate them all).

To me, this looks like a use case for a `debug:feature-toggles` command allowing to show the configuration of toggles (or maybe `debug:container --tag="feature_toggle.feature"` would already be enough)

Reply:
a dedicated command could be indeed the solution. To me the debug is not enough as it doesn't show each inner configuration and nested strategies.